src/test/java/de/tum/in/www1/artemis/service/ConsistencyCheckServiceTest.java

```diff
@@ -0,0 +1,174 @@
+        var exercise = (ProgrammingExercise) course.getExercises().iterator().next();
+        exercise.setTestRepositoryUrl("artemislocalhost/to/set/localSimulation/to/true");
+        exercise = programmingExerciseRepository.save(exercise);
+
+        var consistencyErrors = request.getList("/api/consistency-check/exercise/" + exercise.getId(), HttpStatus.OK, ConsistencyErrorDTO.class);
+        assertThat(consistencyErrors.size()).isEqualTo(1);
+        assertThat(consistencyErrors.get(0).getType()).isEqualTo(ConsistencyErrorDTO.ErrorType.IS_LOCAL_SIMULATION);
+    }
+
+    /**
+     * Test consistencyCheck feature with a course
+     * containing errors
+     * @throws Exception if an error occurs
+     */
+    public void checkConsistencyOfCourse() throws Exception {
+        var newExercise = ModelFactory.generateProgrammingExercise(null, null, course);
+        newExercise.setShortName("Test2");
+        newExercise.setTitle("Test2");
+        newExercise = programmingExerciseRepository.save(newExercise);
+        database.addTemplateParticipationForProgrammingExercise(newExercise);
+        database.addSolutionParticipationForProgrammingExercise(newExercise);
+        newExercise = programmingExerciseRepository.save(newExercise);
+        course.addExercises(newExercise);
+
+        Iterator<Exercise> iterator = course.getExercises().iterator();
+        var exercise1 = (ProgrammingExercise) iterator.next();
+        var exercise2 = (ProgrammingExercise) iterator.next();
+        exercise1 = programmingExerciseRepository.findByIdWithTemplateAndSolutionParticipationElseThrow(exercise1.getId());
+        exercise2 = programmingExerciseRepository.findByIdWithTemplateAndSolutionParticipationElseThrow(exercise2.getId());
+
+        mockDelegate.mockCheckIfProjectExistsInVcs(exercise1, false);
+        mockDelegate.mockCheckIfBuildPlanExists(exercise1.getProjectKey(), exercise1.getTemplateBuildPlanId(), false, false);
+        mockDelegate.mockCheckIfBuildPlanExists(exercise1.getProjectKey(), exercise1.getSolutionBuildPlanId(), true, false);
+
+        mockDelegate.mockCheckIfProjectExistsInVcs(exercise2, true);
+        mockDelegate.mockRepositoryUrlIsValid(exercise2.getVcsTemplateRepositoryUrl(), exercise2.getProjectKey(), true);
+        mockDelegate.mockRepositoryUrlIsValid(exercise2.getVcsTestRepositoryUrl(), exercise2.getProjectKey(), true);
+        mockDelegate.mockRepositoryUrlIsValid(exercise2.getVcsSolutionRepositoryUrl(), exercise2.getProjectKey(), true);
+        mockDelegate.mockCheckIfBuildPlanExists(exercise2.getProjectKey(), exercise2.getTemplateBuildPlanId(), true, false);
+        mockDelegate.mockCheckIfBuildPlanExists(exercise2.getProjectKey(), exercise2.getSolutionBuildPlanId(), false, false);
+        var consistencyErrors = request.getList("/api/consistency-check/course/" + course.getId(), HttpStatus.OK, ConsistencyErrorDTO.class);
+        assertThat(consistencyErrors.size()).isEqualTo(3);
```

Comment:
Please check as well that the correct errors are returned for the correct exercises.